projects/ngneat/transloco/src/lib/transloco.service.ts

```diff
@@ -104,6 +104,7 @@ export class TranslocoService implements OnDestroy {
 
   setActiveLang(lang: string) {
     this.lang.next(lang);
+    this.parser.onLangChanged && this.parser.onLangChanged(lang);
```

Comment:
@itayod 
It must be done before `this.lang.next(lang)` ?
If not, in my project, messages with currency are formatted in previous active lang.

Reply:
If you have an issue you can open a new bug with a reproduction link, this is from 2 years ago 🙂